src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx

```diff
@@ -151,264 +221,146 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
   //     We should handle this in the UI and block users from deposits that *will* fail, by detecting their current position(s)
   //     and not allowing them to select the sure-to-fail deposit types
   const defaultOpportunityId = useMemo(() => {
-    if (!parsedPools) return undefined
-    if (opportunityId) return undefined
-    if (paramOpportunityId) return paramOpportunityId
+    if (!pools?.length) return
 
-    // This may be wrong if the asset doesn't support said opportunity in an asym way, but this is only used to indicate the default pool really
-    // Trying to fix this to be semantically correct will result in circular dependencies madness, don't try this if you value your sanity
-    const firstAsymOpportunityId = parsedPools.find(pool => pool.asymSide === null)?.opportunityId
+    const assetId = poolAssetIdToAssetId(poolAssetId ?? '')
 
-    return firstAsymOpportunityId
-  }, [parsedPools, opportunityId, paramOpportunityId])
+    const walletSupportedOpportunity = pools.find(pool => {
+      const { chainId } = fromAssetId(pool.assetId)
+      return walletSupportsChain({ chainId, wallet, isSnapInstalled })
+    })
 
-  const [poolAssetAccountAddress, setPoolAssetAccountAddress] = useState<string | undefined>(
-    undefined,
-  )
-  const [activeOpportunityId, setActiveOpportunityId] = useState(
-    opportunityId ?? defaultOpportunityId,
-  )
+    const opportunityType = getDefaultOpportunityType(
+      assetId || walletSupportedOpportunity?.assetId || pools[0].assetId,
+    )
 
-  const { data: isSmartContractAccountAddress, isLoading: isSmartContractAccountAddressLoading } =
-    useIsSmartContractAddress(poolAssetAccountAddress ?? '')
+    const poolOpportunityId = assetId ? `${assetId}*${opportunityType}` : ''
```

Comment:
Perhaps use new util here and on the one underneath too?

Reply:
https://github.com/shapeshift/web/pull/6278/commits/5c8410fdedc3db1e4c41560deca214f6ce72b45c